src/core/featurelistmodel.h

```diff
@@ -20,12 +20,162 @@
 
 #include <QAbstractItemModel>
 #include <QTimer>
+#include <QThread>
+#include <QMutex>
 
+#include "qgsapplication.h"
+#include "qgslogger.h"
+#include "qgsvectorlayer.h"
+#include "qgsvectorlayerfeatureiterator.h"
 #include <qgsfeature.h>
 #include <qgsstringutils.h>
 
 class QgsVectorLayer;
 
+/**
+ * \class QgsFieldExpressionValuesGatherer
+ * Gathers features with substring matching on an expression.
```

Comment:
could you add a not that this class is an exact copy of a QGIS internal one and that we should try to bring any modification to both files?

Reply:
and probably move this to a dedicated file